numba/tests/doc_examples/test_examples.py

```diff
@@ -220,6 +354,204 @@ def g(x, y, res):
             self.assertIsInstance(result, np.integer)
             self.assertEqual(result, 20)
 
+    def test_guvectorize_overwrite(self):
+        with captured_stdout():
+            # magictoken.ex_guvectorize_overwrite.begin
+            from numba import guvectorize, float64
+            import numpy as np
+
+            @guvectorize([(float64[:], float64[:])], '()->()')
+            def init_values(invals, outvals):
+                invals[0] = 6.5
+                outvals[0] = 4.2
+            # magictoken.ex_guvectorize_overwrite.end
+
+            # magictoken.ex_guvectorize_overwrite_call_one.begin
+            invals = np.zeros(shape=(3, 3), dtype=np.float64)
+            # invals == array([[6.5, 6.5, 6.5],
+            #                  [6.5, 6.5, 6.5],
+            #                  [6.5, 6.5, 6.5]])
+
+            outvals = init_values(invals)
+            # outvals == array([[4.2, 4.2, 4.2],
+            #                   [4.2, 4.2, 4.2],
+            #                   [4.2, 4.2, 4.2]])
+            # magictoken.ex_guvectorize_overwrite_call_one.end
+
+            self.assertIsInstance(invals, np.ndarray)
+            correct = np.array([
+                [6.5, 6.5, 6.5],
+                [6.5, 6.5, 6.5],
+                [6.5, 6.5, 6.5]])
+            np.testing.assert_array_equal(invals, correct)
+
+            self.assertIsInstance(outvals, np.ndarray)
+            correct = np.array([
+                [4.2, 4.2, 4.2],
+                [4.2, 4.2, 4.2],
+                [4.2, 4.2, 4.2]])
+            np.testing.assert_array_equal(outvals, correct)
+
+            # magictoken.ex_guvectorize_overwrite_call_two.begin
+            invals = np.zeros(shape=(3, 3), dtype=np.float32)
+            # invals == array([[0., 0., 0.],
+            #                  [0., 0., 0.],
+            #                  [0., 0., 0.]], dtype=float32)
+            outvals = init_values(invals)
+            # outvals == array([[4.2, 4.2, 4.2],
+            #                   [4.2, 4.2, 4.2],
+            #                   [4.2, 4.2, 4.2]])
```

Comment:
I would suggest also showing `invals` is still all zeros here - it's no longer clear that this is the case - in the original text, `invals` was printed out after the call to `init_values()`.